### PR TITLE
feat: [CI-3676] support input set handling

### PR DESCRIPTION
**Pull request by harshjain12**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31230)

### CODE REVIEW

File: 320-ci-execution/src/main/java/io/harness/ci/integrationstage/CIStepGroupUtils.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ private ExecutionWrapperConfig getGitCloneStep(
     if (ciCodebase == null) {
       throw new CIStageExecutionException("Codebase is mandatory with enabled cloneCodebase flag");
     }
-    Integer depth = ciCodebase.getDepth();
+    Integer depth = ciCodebase.getDepth().getValue();
```
Comment (shubham149) [inline 320-ci-execution/src/main/java/io/harness/ci/integrationstage/CIStepGroupUtils.java:219]: Use io.harness.beans.serializer.RunTimeInputHandler#resolveIntegerParameter

File: 320-ci-execution/src/main/java/io/harness/ci/integrationstage/CIStepGroupUtils.java [hunk 2]
```diff
@@ -231,13 +231,13 @@ private ExecutionWrapperConfig getGitCloneStep(
...
-    if (ciCodebase.getPrCloneStrategy() != null) {
+    if (ciCodebase.getPrCloneStrategy().getValue() != null) {
       settings.put(PR_CLONE_STRATEGY_ATTRIBUTE,
-          JsonNodeFactory.instance.textNode(ciCodebase.getPrCloneStrategy().getYamlName()));
+          JsonNodeFactory.instance.textNode(ciCodebase.getPrCloneStrategy().getValue().getYamlName()));
     }
 
     Map<String, String> envVariables = new HashMap<>();
-    if (ciCodebase.getSslVerify() != null && !ciCodebase.getSslVerify()) {
+    if (ciCodebase.getSslVerify().getValue() != null && !ciCodebase.getSslVerify().getValue()) {
```
[the remark this answers is not in the document]
Comment (shubham149) [reply 320-ci-execution/src/main/java/io/harness/ci/integrationstage/CIStepGroupUtils.java:240]: Check all the places where we use getValue, are we doing null checks correctly or not?

File: 889-yaml-commons/src/main/java/io/harness/yaml/extended/ci/validator/ResourceValidatorConstants.java [hunk 3]
```diff
@@ -13,5 +13,5 @@
 
 @OwnedBy(CI)
 public interface ResourceValidatorConstants {
-  String MEMORY_PATTERN = "^(([0-9]*[.])?[0-9]+)([GM]i?)$";
+  String MEMORY_PATTERN = "^(([0-9]*[.])?[0-9]+)([GM]i?)|<\\+input>$";
```
[the remark this answers is not in the document]
Comment (harshjain12) [reply 889-yaml-commons/src/main/java/io/harness/yaml/extended/ci/validator/ResourceValidatorConstants.java:16]: It was matching. I tried that but will test it again

**Review by shubham149 — COMMENTED** [review]
> Reviewed 28 of 30 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31230)* status: 0 of 1 approvals obtained.

**Review by shubham149 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31230#-:-MxdqHr92AzW9ZamUl3P:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**